plugins/outputs/graphite/graphite.go

```diff
@@ -71,6 +72,31 @@ func (g *Graphite) Description() string {
 	return "Configuration for Graphite server to send metrics to"
 }
 
+// We need check eof as we can write to nothing without noticing anything is wrong
+// the connection stays in a close_wait
+// We can detect that by finding an eof
+// if not for this, we can happily write and flush without getting errors (in Go) but getting RST tcp packets back (!)
+// props to Tv via the authors of carbon-relay-ng` for this trick.
+func checkEOF(conn *net.Conn) {
+	b := make([]byte, 1024)
+	(*conn).SetReadDeadline(time.Now().Add(10 * time.Millisecond))
```

Comment:
why are you dereferencing `conn` every time you make a function call? I don't think this is necessary unless I'm missing something. ie, change `(*conn)` to `conn`